components/resources/quizzes/langQuiz/LangResults.tsx

```diff
@@ -3,22 +3,70 @@ import SkillifyNavbar from "../shared/SkillifyNavbar";
 
 type LangResultsProps = {
   onBackClick: () => void;
+  score: { [key: string]: number };
 };
 
-const LangResults = ({ onBackClick }: LangResultsProps) => {
+const LangResults = ({ onBackClick, score }: LangResultsProps) => {
+  const maxLangScore = (score: { [key: string]: number }) => {
+    let maxScore = 0;
+    let preferredLang = "";
+    for (const [key, value] of Object.entries(score)) {
+      if (value > maxScore) {
+        maxScore = value;
+        preferredLang = key;
+      }
+    }
+    return preferredLang;
+  };
+
+  const preferredLang = maxLangScore(score);
+
+  type resultData = {
```

Comment:
types should be capitalized. Use a more descriptive name `FirstLanguageQuizResultData`